nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+</pre>
+
+
+'''Success return value'''
+
+result: Object - Response from RPC node
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_REFUSED`|`RPC_ERROR`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+invokeRead({
+  scriptHash: '505663a29d83663a838eee091249abd167e928f5',
+  operation: 'calculatorAdd',
+  arguments: [
+    {
+      type: 'integer',
+      value: 2
+    },
+    {
+      type: 'integer',
+      value: 10
+    }
+  ]
+})
+.then((result: Object) => {
+  console.log('Read invocation result: ' + JSON.stringigy(result));
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_REFUSED:
+      console.log('Connection dApp not connected. Please call the "connect" function.');
+      break;
+   case RPC_ERROR:
+    console.log('There was an error when broadcasting this transaction to the network.');
+    break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request
+* Validate and format the inputs
+** Script hash is valid
+** Format parameters
+*** If `byteArray`, convert to appropriate hexstring based on assumed format
+* Broadcast the RPC request
+* Return the RPC response results
+
+Example
+<pre>
+{
+  script: '8h89fh398f42f.....89hf2894hf9834',
+  state: 'HALT, BREAK',
+  gas_consumed: '0.13',
+  stack: [
+    {
+      type: 'Integer',
+      value: '1337'
+    }
+  ]
+}
+</pre>
+
+
+====getBlock====
+Get information about a specific block.
+
+
+'''Method Interface'''
+
+<pre>
+function getBlock({
+  blockHeight: integer,
+  network?: string
+}): Promise<{result: BlockDetails}>
+</pre>
+
+
+'''Input arguments'''
+
+<pre>
+interface GetBlockInputArgs {
+  blockHeight: integer;
+  network?: string;
+}
+</pre>
+
+
+'''Success return value'''
+
+<pre>
+interface BlockDetails {
+  hash: string; // Block hash
+  size: number; // Block size (bytes)
+  version: number; // The version number of the block execution
+  previousblockhash: string; // Previous block Hash
+  merkleroot: string; // Merkel root
+  time: number; // Block generation timestamp
+  index: number; // Block index (height)
+  nonce: string; // Block pseudo-random number
+  nextconsensus: string; // Next master biller
+  script: ScriptDetails; // Block call signature authentication information
+  tx: BlockTransactionDetails[]; // Block containing trading group
+  confirmations: number; // Confirmation number (number of blocks after this block)
+  nextblockhash: string; // Next block hash
+}
+
+interface BlockTransactionDetails {
+  txid: string;
+  size: number;
+  type: string;
+  version: number;
+  attributes: any[];
+  vin: any[];
+  vout: any[];
+  sys_fee: string;
+  net_fee: string;
+  scripts: any[];
+  nonce: number;
+}
+
+interface ScriptDetails {
```

Comment:
ScriptDetails is called Witness on Neo2/Neo3 (don't know for ONT).

Reply:
There was no mention of the term Witness in the response documentation, so it was left as a reference to the `script` key.

Reply:
Can it be changed? I feel like propagating an error here.